### PR TITLE
Converted to VS2017 and added support for .Net Standard 2.0 and snupkg

### PR DESCRIPTION
**Pull request by snakefoot**

### CODE REVIEW

**Comment by snakefoot** [comment]
@304NotModified Finally ready for review. Have a bonus PR that makes it possible to create EventSource with dynamic ProviderName (Instead of all being called "LowLevelDesign-NLogEtwSource", terrible default name btw.).

File: NLog.Etw/NLogEtwExtendedTarget.cs [hunk 1]
```diff
@@ -56,31 +61,45 @@ public void Critical(string LoggerName, string Message)
...
+                {
+                    if (EtwLogger.Log.IsEnabled(EventLevel.Error, EventKeywords.None))
+                    {
+                        var message = Layout.Render(logEvent);
+                        EtwLogger.Log.Error(logEvent.LoggerName, message);
+                    }
+                }
+                else //if (logEvent.Level == LogLevel.Fatal)
+                {
+                    var message = Layout.Render(logEvent);
```
Comment (304NotModified) [inline NLog.Etw/NLogEtwExtendedTarget.cs:100]: Critical can't be disabled?

**Review by 304NotModified — APPROVED** [review]

**Comment by 304NotModified** [comment]
Thanks! Great addition!

**Comment by snakefoot** [comment]
Bonus  PR #19 is ready.

**Comment by snakefoot** [comment]
@304NotModified Should also be included in the 4.2-release-version.

**Comment by 304NotModified** [comment]
Thanks!

**Pull request merged**